### ISSUE TITLE
Is it possible to run it through the CPU if there is no GPU? Will the accuracy be affected?

### ISSUE THREAD

**Issue by shuangshunji** [tencent-ailab/Interformer#5]
4. Giving the docking poses, predicting its PoseScore and Affinity values.

PYTHONPATH=interformer/ python inference.py -test_csv examples/demo_dock.round0.csv \
-work_path examples/ \
-ligand_folder infer/ \
-ensemble checkpoints/v0.2_affinity_model/model* \
-gpus 1 \

**Comment by lkfo415579** [comment]
Accuracy will not be affected, you may change the device of the model to CPU.

**Comment by shuangshunji** [comment]
change【 -gpus 1  】to  【-cpus 1  】？

**Comment by shuangshunji** [comment]
Does 1 represent a thread?Would changing the cpus from 1 to 2 or more increase the accuracy of the prediction?

**Comment by lkfo415579** [comment]
@shuangshunji  No, there is no cpus parameters. You have to modify the code of inference.py, assign a cpu device to the model.

**Comment by shuangshunji** [comment]
Thank you for your busy reply! I would also like to ask, if I want to use Interformer to dock my target protein and small molecule ligand, at which step can I get my docking file, is it the second step（2. Predicting energy functions file. Download checkpoints from [zenodo](https://zenodo.org/doi/10.5281/zenodo.10828798).） or the third step（3.Giving the energy files produce docking poses via MonteCarlo sampling.）?Which folder contains the results of the docking file？Thanks a lot ！

**Comment by lkfo415579** [comment]
The third step will produce the docking poses. And the docking file is located in ligand_reconstructing/ folder.

**Comment by shuangshunji** [comment]
Thanks for the pointers and congratulations again on a job well done!

**Comment by shuangshunji** [comment]
The ligand_reconstructing/ folder contains a file named  _docked.sdf, When I opened it with the visualization software, it showed only the ligand, not the protein
![6981734435884_ pic](https://github.com/user-attachments/assets/a6ea9a24-968f-4648-802c-4b1339a4694c)

**Comment by lkfo415579** [comment]
@shuangshunji Most protein-ligand docking programs only generate small-molecule SDF files as output. You have to combine the protein structure ($PDB_pocket.pdb) to see the result. I recommend using `PyMol` software to do the visualization. It is fast and free, and most people in the field are using it.

**Comment by shuangshunji** [comment]
Could you explain in detail how to get PDB_docked.sdf to bind $PDB_pocket.pdb  to get a complete protein-ligand docking file to further analyze the amino acid-ligand interactions?

**Comment by shuangshunji** [comment]
Higher values of PoseScore and Affinity values represent better binding models？

**Comment by lkfo415579** [comment]
@shuangshunji Yes. You may sort by the pose score first then select those high affinity values molecule candidates.

**Comment by shuangshunji** [comment]
Question about resoult/ demo_dock.round0_ensemnle.csv .There is a column that reads energy .Is it Bingding Energy（kcal/mol）?

**Comment by lkfo415579** [comment]
The score value indicates the likelihood of a correct docking pose, with higher values representing greater likelihood. It is important to note that this unit is not equivalent to kcal/mol.

**Comment by shuangshunji** [comment]
I mean this column.
![image](https://github.com/user-attachments/assets/31658a0a-bb08-4ae7-b888-cd0d881366f7)

**Comment by lkfo415579** [comment]
@shuangshunji The score value indicates the likelihood of a correct docking pose, with lower values representing greater likelihood. It is important to note that this unit is not equivalent to kcal/mol. Since the energy score means summing up all the protein-ligand atom pairs together and then multiplying the result by -1, minimising instead of maximising a score value is much simpler in the sampling process.

**Comment by shuangshunji** [comment]
Docking predictions using different ligand molecules and the same protein, and whether their affinity values(pre_pIC50) can be used as evidence to prove which is more affine.

**Comment by lkfo415579** [comment]
@shuangshunji Yes, it can. It is even comparable in different protein systems.

**Comment by shuangshunji** [comment]
thanks a lot

**Comment by shuangshunji** [comment]
After running the program, I successfully obtained 21 candidate docking ligands. Should I refer to pred_pose or rmsd to select the best one?

**Comment by shuangshunji** [comment]
When I used the crystal structure as a test, I found that the rmsd=0 value given by the Interformer was exactly the ligand position of the crystal structure. So, is rmsd the best selection basis for candidate ligands, rather than pred_pose?

**Comment by lkfo415579** [comment]
@shuangshunji  In practice, the RMSD (Root Mean Square Deviation) is calculated by comparing the predicted pose to a reference ligand. However, in real-world scenarios, this reference value is unavailable. (It is used for testing benchmark only, for convenience) It is recommended to use the `pred_pose` as the selection criterion for selecting the docking poses.
